src/editor/classes/AppManager.js

```diff
@@ -0,0 +1,121 @@
+//"use strict";
+/**
+ * @brief This is our controller class.  All calls are made through here
+ *        so we have a consistent interface for javascript communication.
+ * 
+ * Some brief examples of how to use this:
+ * App.proxy.sendEvent(Message, Data);
+ * App.eventHook.onChange();
+ *
+ */
+var editor;
+class AppManager {
+    constructor()
+    {
+        // Sub-function objects
+        this.proxy = new CommunicationsManager();
+        this.events = new EditorEventHook();
+        this.content = new ContentManager();
+ 
+        //Variables
+        this.defaultTheme = "default";
+        this.loadLink = undefined;
+
+        function getQueryValue(name) 
+        {
+            name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+            let regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
+                results = regex.exec(location.search);
+            if (results !== null) {
+                return decodeURIComponent(results[1].replace(/\+/g, " "));
+            }
+            return "";
+        }
+
+        //Try to get the current preferred theme.
+        let themePath = getQueryValue("themePath");
+        let themeName = getQueryValue("themeName");
+        if (themePath !== "") {
+            this.setEditorStyleSheet(themePath);
+            this.defaultTheme = themeName;
+        }
+        $(document).ready(() => {this.initializeCodeMirror()});
+    }
+
+    setEditorStyleSheet(sheet)
+    {
```

Comment:
It seems that this method doesn't just set a stylesheet, but actually changes the theme of CodeMirror (adding a stylesheet as a side effect).

A better name would be setTheme, with a comment explaining that the associated stylesheet will be loaded at runtime.